elastic/kibana
# [Discover] Fix search sessions using temporary data views

## Summary

This PR takes care that search sessions in Discover work correctly with temporary data view

### Testing
1. Go to Discover
2. Enter a valid index pattern in the data view picker
3. Click on 'Explorer {x} indices' ... this will create a temporary data view behind the scenes
4. Once the search is done, click on the checkmark next in Discover's breadcrump
5. Click on "Save Session"
6. Clip on "Manage Sessions"
7. When the status of your search in the Search Sessions UI is completed, click on your previously saved search session
8. This should restore your previous search in Discover 

Before this PR this didn't work for temporary data views

Fixes #156675


### Checklist
